integration/raft/channel_participation_test.go

```diff
@@ -365,6 +369,124 @@ var _ = Describe("ChannelParticipation", func() {
 			channelInfo := channelparticipation.ListOne(network, orderer3, "participation-trophy")
 			Expect(channelInfo).To(Equal(expectedChannelInfoPT))
 		})
+
+		It("join application channel with join-block as follower via channel participation api", func() {
+			orderer1 := network.Orderer("orderer1")
+			orderer2 := network.Orderer("orderer2")
+			orderer3 := network.Orderer("orderer3")
+			orderers := []*nwo.Orderer{orderer1, orderer2}
+			members := []*nwo.Orderer{orderer1, orderer2}
+			peer := network.Peer("Org1", "peer0")
+
+			By("starting two orderers")
+			for _, o := range orderers {
+				startOrderer(o)
+				channelparticipation.List(network, o, nil)
+			}
+
+			genesisBlock := applicationChannelGenesisBlock(network, members, peer, "participation-trophy")
+			expectedChannelInfoPT := channelparticipation.ChannelInfo{
+				Name:            "participation-trophy",
+				URL:             "/participation/v1/channels/participation-trophy",
+				Status:          "active",
+				ClusterRelation: "member",
+				Height:          1,
+			}
+
+			for _, o := range members {
+				By("joining " + o.Name + " to channel as a member")
+				channelparticipation.Join(network, o, "participation-trophy", genesisBlock, expectedChannelInfoPT)
+				channelInfo := channelparticipation.ListOne(network, o, "participation-trophy")
+				Expect(channelInfo).To(Equal(expectedChannelInfoPT))
+			}
+
+			By("waiting for the leader to be ready")
+			memberRunners := []*ginkgomon.Runner{ordererRunners[0], ordererRunners[1]}
+			leader := findLeader(memberRunners)
+
+			By("ensuring the channel is usable by submitting a transaction to each member")
+			env := CreateBroadcastEnvelope(network, peer, "participation-trophy", []byte("hello"))
+			for i, o := range members {
+				By("submitting transaction to " + o.Name)
+				expectedChannelInfoPT.Height++
+				i++
```

Comment:
a nit again, but I found it's very weird you are using slice iteration index to assert for certain block number reception, instead of having clearly readable and explicit variable for that.

Reply:
Agreed. This does seem weird to use "i" here. 

Maybe we should change this and the similar loops elsewhere in this test file to 
1) create a test helper that does the logic currently in the loop to submit a transaction and check the block number and channel info return by List. It can take the expected channel info and expected block number as arguments.
2) use the helper function directly for each orderer we're trying to test instead of looping (aka 'testHelper(members[0]...)` that way we explicitly set the expected channel info for each and directly pass the expected block number.

I know I'm responsible for the loop logic that existed before this PR so let me know if you'd like me to clean things up. :)

Reply:
I revised using something similar to the 2nd option @wlahti proposed for this test